packages/frameworks/src/frameworks.ts

```diff
@@ -277,8 +277,9 @@ export const frameworks = [
         handle: 'filesystem',
       },
       {
-        src: '/(.*)',
-        dest: '/index.html',
```

Comment:
This will break existing astro deployments that use SPA pattern.

See https://vercel.com/guides/custom-404-page

Reply:
@styfle Ah. I didn't know that Astro outputs SPA pages. I thought it would always a file for each page (I was reading the [docs here](https://docs.astro.build/en/core-concepts/astro-pages/) but I'm not really familiar with Astro).

Reply:
> Yeah, we need to explicitly specify that only 404 statuses can return a 404 error page, but other redirects work as expected

Aren't all Astro outputs static? How would these static assets return 404 statuses?

Reply:
Yeah but that's only if you're using Astro in static mode, but Astro also support SSR mode, I think that what @styfle is referencing

Reply:
Can we detect SPA vs SSG mode, and change the routes dynamically based on astro config?

Reply:
@styfle How do you use Astro in SPA mode?

Reply:
Call it SSR mode not SPA mode, it sounds confusing when it's called SPA mode, as Astro doesn't have an official SPA mode. To answer your question @styfle yes you should be able to, I know that the Astro team has discussed adding some sort of mode value to determine what mode is being used.

Reply:
@lucleray I'm not sure, I haven't used Astro.

This route was added by @tony-sull in the original PR to add Astro support https://github.com/vercel/vercel/pull/7747

Presumably that route was necessary for the Astro example/demo that was added there too.

Reply:
Ok, I've got confirmation it seems we're supposed to use `import.meta.env.SSR` to determine if Astro is running in SSR or SSG mode

Reply:
> Yeah but that's only if you're using Astro in static mode, Astro also support SSR mode, I think that what @styfle is referencing.

If Astro is used in SSR mode, I think that it's Astro itself (the Astro "server") that should render the 404.html page and return a 404 status when the page is not available. 

I don't think that we currently support Astro in SSR mode (at least not with this configuration - it might work with the Astro adapter and `vc build`).

Reply:
Squash that, `import.meta.env.SSR` is currently buggy, I'll keep doing some investigation

Reply:
The more I think about it, @lucleray I think your solution here should work for Astro as there is no SPA mode for Astro. However, it may break other frameworks

Reply:
Good news everybody Astro has merged a fix from their end https://github.com/withastro/astro/pull/4018#event-7048931698

Reply:
Astro doesn't conventionally do SPA's by default

> @delucis : Astro doesn’t really do SPA builds. I mean, they’re technically possible if you just have src/pages/index.astro and mount a client-side routed app there, but it’s MPA as a basic premise both in static and SSR modes.

Is there a way to enable this pr while not harming other SPA frameworks?